code/src/mlx90640.cpp

```diff
@@ -0,0 +1,64 @@
+#include <mlx90640.hpp>
+
+namespace r2d2::thermal_camera {
+
+    mlx90640_c::mlx90640_c(i2c::i2c_bus_c &bus, const uint8_t address)
+        : mlx_i2c_bus(bus, address), mlx_processor(mlx_i2c_bus) {
+    }
+
+    void mlx90640_c::toggle_nth_bit(uint16_t &source, const uint8_t n,
+                                    const uint8_t to) const {
+        source = (source & ~(1U << n)) | (to << n);
+    }
+
+    uint16_t mlx90640_c::get_refresh_rate() const {
+        uint16_t rate = mlx_i2c_bus.read_register(INTERNAL_CONTROL_REGISTER);
+        rate >>= 7 & 7;
+        return 1 << (rate - 1);
+    }
+
+    void mlx90640_c::set_refresh_rate(uint16_t refresh_rate) const {
+        if (refresh_rate > MAX_REFRESH_RATE) {
```

Comment:
Why do you return, instead of clamping and setting it to the max refresh rate?

Reply:
FTFY